catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -774,6 +778,27 @@ FileSystem getFileSystem(Path path, Map<String, String> config) throws IOExcepti
               scheme, path, fileSystemProvidersMap.keySet(), fileSystemProvidersMap.values()));
     }
 
-    return provider.getFileSystem(path, config);
+    int timeoutSeconds =
+        (int)
+            propertiesMetadata
+                .catalogPropertiesMetadata()
+                .getOrDefault(config, HadoopCatalogPropertiesMetadata.GET_FILESYSTEM_TIMEOUT_SECONDS);
+    try {
+      AtomicReference<FileSystem> fileSystem = new AtomicReference<>();
+      Awaitility.await()
+          .atMost(timeoutSeconds, TimeUnit.SECONDS)
+          .until(
+              () -> {
+                fileSystem.set(provider.getFileSystem(path, config));
```

Comment:
Why it is so time-consuming to initialize the filesystem client?

Reply:
If the user sets an incorrect endpoint, the client will retry to get the connection for a certain amount of time.

Reply:
I don't think you really fix this problem without using another thread to create a FS and polling the status asynchronously?